File: datasets/php-keep-code_davinci_001_temp_0.2/HumanEval_25_factorize.php
<?php
// Return list of prime factors of given integer in the order from smallest to largest.
// Each of the factors should be listed number of times corresponding to how many times it appeares in factorization.
// Input number should be equal to the product of all factors
// >>> factorize(8)
// [2, 2, 2]
// >>> factorize(25)
// [5, 5]
// >>> factorize(70)
// [2, 5, 7]
function factorize($n) {
    $result = [];
    $i = 2;
    while ($n > 1) {
        while ($n % $i == 0) {
            $result[] = $i;
            $n /= $i;
        }
        $i++;
    }
    return $result;
}


function candidate(...$args) {
    return factorize(...$args);
}

function test(): void {
    if (candidate(2) !== array(2)) { throw new Exception("Test failed!"); }
    if (candidate(4) !== array(2, 2)) { throw new Exception("Test failed!"); }
    if (candidate(8) !== array(2, 2, 2)) { throw new Exception("Test failed!"); }
    if (candidate(57) !== array(3, 19)) { throw new Exception("Test failed!"); }
    if (candidate(3249) !== array(3, 3, 19, 19)) { throw new Exception("Test failed!"); }
    if (candidate(185193) !== array(3, 3, 3, 19, 19, 19)) { throw new Exception("Test failed!"); }
    if (candidate(20577) !== array(3, 19, 19, 19)) { throw new Exception("Test failed!"); }
    if (candidate(18) !== array(2, 3, 3)) { throw new Exception("Test failed!"); }
}

test();

File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_25_factorize.lua
-- Return list of prime factors of given integer in the order from smallest to largest.
-- Each of the factors should be listed number of times corresponding to how many times it appeares in factorization.
-- Input number should be equal to the product of all factors
-- >>> factorize(8)
-- [2, 2, 2]
-- >>> factorize(25)
-- [5, 5]
-- >>> factorize(70)
-- [2, 5, 7]
local function factorize(n)
    local factors = {}
    local factor = 1
    local i = 1
    while i <= n do
        if n % i == 0 then
            factor = factor * i
            i = i + 1
        else
            i = i + 1
        end
    end
    factors[#factors + 1] = factor
    return factors
end

-- Return list of prime factors of given integer in the order from smallest to largest.
-- Each of the factors should be listed number of times corresponding to how many times it appeares in factorization.
-- Input number should be equal to the product of all factors
-- >>> factorize(8)
-- [2, 2, 2]
-- >>> factorize(25)
-- [5, 5]
-- >>> factorize(70)
-- [2, 5, 7]

lu = require('luaunit')

function test_humaneval()
local candidate = factorize
    lu.assertEquals(candidate(2), {2})
    lu.assertEquals(candidate(4), {2, 2})
    lu.assertEquals(candidate(8), {2, 2, 2})
    lu.assertEquals(candidate(57), {3, 19})
    lu.assertEquals(candidate(3249), {3, 3, 19, 19})
    lu.assertEquals(candidate(185193), {3, 3, 3, 19, 19, 19})
    lu.assertEquals(candidate(20577), {3, 19, 19, 19})
    lu.assertEquals(candidate(18), {2, 3, 3})
end

os.exit(lu.LuaUnit.run())